pydough/metadata/properties/masked_table_column_metadata.py

```diff
@@ -0,0 +1,189 @@
+        comp: list = super().components
+        comp.append(self.unprotect_protocol)
+        comp.append(self.protect_protocol)
+        comp.append(self.server_masked)
+        return comp
+
+    @staticmethod
+    def parse_from_json(
+        collection: CollectionMetadata, property_name: str, property_json: dict
+    ) -> None:
+        """
+        Procedure dispatched from PropertyMetadata.parse_from_json to handle
+        the parsing for table column properties.
+
+        Args:
+            `collection`: the metadata for the PyDough collection that the
+            property would be inserted into.
+            `property_name`: the name of the property that would be inserted.
+            `property_json`: the JSON object that would be parsed to create
+            the new table column property.
+
+        Raises:
+            `PyDoughMetadataException`: if the JSON for the property is
+            malformed.
+        """
+        error_name: str = MaskedTableColumnMetadata.create_error_name(
+            property_name, collection.error_name
+        )
+        # Extract the `data_type` and `column_name` fields from the JSON object
+        type_string: str = extract_string(property_json, "data type", error_name)
+        try:
+            data_type: PyDoughType = parse_type_from_string(type_string)
+        except PyDoughTypeException as e:
+            raise PyDoughMetadataException(*e.args)
+        column_name: str = extract_string(property_json, "column name", error_name)
+
+        # Extract the `unprotect protocol`, `protect protocol`, and
+        # `server masked` fields from the JSON object.
+        unprotect_protocol: str = extract_string(
+            property_json, "unprotect protocol", error_name
+        )
+        protect_protocol: str = extract_string(
+            property_json, "protect protocol", error_name
+        )
+        server_masked: bool = False
+        if "server masked" in property_json:
+            server_masked = extract_bool(property_json, "server masked", error_name)
+
+        NoExtraKeys(MaskedTableColumnMetadata.allowed_fields).verify(
+            property_json, error_name
+        )
+
+        # Extract the optional fields from the JSON object.
+        sample_values: list | None = None
+        description: str | None = None
+        synonyms: list[str] | None = None
+        extra_semantic_info: dict | None = None
+        if "sample values" in property_json:
+            sample_values = extract_array(property_json, "sample values", error_name)
+        if "description" in property_json:
+            description = extract_string(property_json, "description", error_name)
+        if "synonyms" in property_json:
+            synonyms = extract_array(property_json, "synonyms", error_name)
+        if "extra semantic info" in property_json:
+            extra_semantic_info = extract_object(
+                property_json, "extra semantic info", error_name
+            )
```

Comment:
This is duplicate of parent class. Instead of duplicating the code, let's move that to be in helper function (`get_optional_fileds` or something)in the parent class and re-use it here?